packages/amplify_datastore/lib/amplify_datastore.dart

```diff
@@ -350,7 +350,9 @@ class NativeAmplifyApi
   Future<NativeGraphQLSubscriptionResponse> subscribe(
       NativeGraphQLRequest request) async {
     final flutterRequest = nativeRequestToGraphQLRequest(request);
-
+    // Turn off then default reconnection behavior to allow native side to trigger reconnect
```

Comment:
Q: I don't see the native code to call WebSocketOptions.autoReconnect = true, is it expected?

Reply:
Yes this is expected. Swift has it's own reconnection logic that is used for DataStore. Having them both enabled created race conditions, so I disabled Flutter's in favor of Swift because it's also tied to restarting the sync engine.